Tests/scripts/configure_tests.py

```diff
@@ -646,14 +685,130 @@ def get_test_from_conf(branch_name):
+    # check if test is skipped
+    if test_id in skipped_tests:
+        return False
+    test_conf = get_test_conf_from_conf(test_id, server_version, conf)
+
+    # check if there's a test to run
+    if not test_conf:
+        return False
+    conf_fromversion = test_conf.get('fromversion', '0')
+    conf_toversion = test_conf.get('toversion', '99.99.99')
+    test_playbooks_set = id_set.get('TestPlaybooks', [])
+    test_playbook_obj = get_obj_from_obj_set(test_id, test_playbooks_set, server_version)
+
+    # check whether the test is runnable in id_set
+    if not test_playbook_obj:
+        return False
+
+    return all([
+        is_test_integrations_available(server_version, test_conf, conf, id_set),    # check used integrations available
+        is_test_scripts_available(test_playbook_obj, server_version, id_set),       # check used scripts available
+        is_runnable_in_server_version(conf_fromversion, server_version, conf_toversion)  # check conf from/to
+    ])
+
+
+def is_test_scripts_available(test_playbook_obj, server_version, id_set):
+    """
+    Check if all used scripts are skipped / available
+    """
+    test_scripts_ids = test_playbook_obj.get('implementing_scripts', [])
+    if test_scripts_ids:
+        if not isinstance(test_scripts_ids, list):
+            test_scripts_ids = [test_scripts_ids]
+        scripts = id_set.get('scripts', [])
+        if any(get_obj_from_obj_set(script_id, scripts, server_version) is None for script_id in
+               test_scripts_ids):
+            return False
+    return True
+
+
+def is_test_integrations_available(server_version, test_conf, conf, id_set):
+    """
+    Check if all used integrations are skipped / available
+    """
+    test_integration_ids = test_conf.get('integrations')
+    if test_integration_ids:
+        if not isinstance(test_integration_ids, list):
+            test_integration_ids = [test_integration_ids]
+        if not is_test_uses_active_integration(test_integration_ids, conf):
+            return False
+        # check integrations from/toversion is valid with server_version
+        integrations_set = id_set.get('integrations', [])
+        if any(get_obj_from_obj_set(integration_id, integrations_set, server_version) is None for integration_id in
+               test_integration_ids):
+            return False
+    return True
+
+
+def is_test_uses_active_integration(integration_ids, conf=None):
+    """Checks whether there's an an integration in test_integration_ids that's not skipped"""
+    if not conf:
+        with open("./Tests/conf.json", 'r') as conf_file:
+            conf = json.load(conf_file)
+    skipped_integrations = conf['skipped_integrations'].keys()
+    # check if all integrations are skipped
+    if all(integration_id in skipped_integrations for integration_id in integration_ids):
+        return False
+    return True
+
+
+def is_any_test_runnable(test_ids, conf=None, id_set=None, server_version='0'):
```

Comment:
not used

Reply:
true, was used, and will be used in a future branch, but I'll just copy it to it